validation-test/stdlib/Collection/LazyFilterCollection.swift.gyb

```diff
@@ -0,0 +1,88 @@
+// -*- swift -*-
+
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %S/../../../utils/gyb %s -o %t/main.swift
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-build-swift %t/main.swift -o %t/LazyFilterCollection.swift.a.out
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-run %t/LazyFilterCollection.swift.a.out
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdlibCollectionUnittest
+
+// Also import modules which are used by StdlibUnittest internally. This
+// workaround is needed to link all required libraries in case we compile
+// StdlibUnittest with -sil-serialize-all.
+import SwiftPrivate
+#if _runtime(_ObjC)
+import ObjectiveC
+#endif
+
+var CollectionTests = TestSuite("Collection")
+
+%{
+variations = [('', 'Sequence'), ('', 'Collection'), ('Bidirectional', 'Collection')]
+}%
+
+// Test collections using value types as elements.
+% for (specifier, kind) in variations:
+CollectionTests.add${specifier}${kind}Tests(
+  make${kind}: { (elements: [OpaqueValue<Int>]) in
+    Minimal${specifier}${kind}(elements: elements).lazy.filter { _ in return true }
+  },
+  wrapValue: identity,
+  extractValue: identity,
+  make${kind}OfEquatable: { (elements: [MinimalEquatableValue]) in
+    Minimal${specifier}${kind}(elements: elements).lazy.filter { _ in return true }
+  },
+  wrapValueIntoEquatable: identityEq,
+  extractValueFromEquatable: identityEq
+)
+% end
+
+// Test collections using reference types as elements.
+% for (specifier, kind) in variations:
+CollectionTests.add${specifier}${kind}Tests(
+  make${kind}: { (elements: [LifetimeTracked]) in
+    Minimal${specifier}${kind}(elements: elements).lazy.filter { _ in return true }
+  },
+  wrapValue: { (element: OpaqueValue<Int>) in
+    LifetimeTracked(element.value, identity: element.identity)
+  },
+  extractValue: { (element: LifetimeTracked) in
+    OpaqueValue(element.value, identity: element.identity)
+  },
+  make${kind}OfEquatable: { (elements: [LifetimeTracked]) in
+    Minimal${specifier}${kind}(elements: elements).lazy.filter { _ in return true }
```

Comment:
I know I asked for a lot of revisions, but this one is very important I think: could you change the closure to be less predictable than always-true?  You would need to inject some extra data into the array, so that you can skip over it.

That is, given input elements `[ 1010, 2020, 3030, 4040 ]` we would inject elements like this: `[ 1010, 2020, 9999, 3030, 9999, 9999, 4040 ]` (just an example) and then adjust the predicate to skip the extra elements.  You can even use a random number generator from `stdlib/private/SwiftPrivate/PRNG.swift` to create different patterns each time.

Why is this important?  There is tricky logic in moving the index forward/backward in lazy filter collection, because we sometimes need to skip over multiple elements of the underlying collection.